### PR TITLE
feat: add and improve components, add new stores 

### LINKED ISSUE #75: 🚀 - add error handler
### Description

Из критериев:

> В проекте отсутствуют необработанные исключения. Все ошибки как с сервера так и с клиента должны выводиться в удобном виде пользователю. 

Предлагаю в качестве отображения использовать ([Snackbar](https://mui.com/material-ui/react-snackbar/#use-with-alerts)) со стилизацией оповещений. Основная сложность - отслеживание всех исключений и их перехват.

### PR DESCRIPTION
**Pull request by TheKenkei**
Closes #75 #74 

add new package: `react-helmet`

improve : `App`, `Maps`, `Sidebar` components
add new components: `Error`, `Loading`

improve  `app` store
add new stores : `title`, `maps`, `links`

delete unused `OtherComponent`

add new styles to `theme.tsx`

chore: refractor code with `prettier`

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -16,37 +18,49 @@ import {
...
+              <Route path="/auth" element={<Auth />}>
+                <Route path="login" index element={<Login />} />
+                <Route path="register" element={<Register />} />
+                <Route path="forgot-password" element={<ForgotPassword />} />
+                <Route path="change-password" element={<ChangePassword />} />
+              </Route>
+            </Routes>
+
+            <Helmet>
+              <meta charSet="utf-8" />
```
Comment (olegnowikow) [inline src/App.tsx:56]: Why is this here?
Charset is defined in `index.html`

File: src/App.tsx [hunk 2]
```diff
@@ -16,37 +18,49 @@ import {
...
+              <title> {titleStore.title} - RouteLink</title>
+              <link rel="canonical" href="http://mysite.com/example" />
```
Comment (olegnowikow) [inline src/App.tsx:58]: same things in `index.html`

**Review by olegnowikow — CHANGES_REQUESTED** [review]
use [notistack](https://notistack.com/features/basic#maximum-snackbars)

**Comment by TheKenkei** [comment]
@olegnowikow the error store stores only one error. Using [notistack](https://notistack.com/features/basic#maximum-snackbars) is impractical

**Review by olegnowikow — APPROVED** [review]

**Pull request merged**